types/tipset_test.go

```diff
@@ -54,220 +49,138 @@ func block(t *testing.T, height int, parentCid cid.Cid, parentWeight uint64, msg
+		assert.Equal(t, uint64(b1.Height), tsHeight)
+		tsTicket, _ := ts.MinTicket()
+		assert.Equal(t, b1.Ticket, tsTicket)
+
+		assert.Equal(t, "{ "+b1.Cid().String()+" }", ts.String())
+	})
 
-func TestTipSetToSortedCidSet(t *testing.T) {
-	tf.UnitTest(t)
+	t.Run("construction of multiblock", func(t *testing.T) {
+		ts := RequireNewTipSet(t, b3, b2, b1) // Presented in reverse order
+		assert.True(t, ts.Defined())
+		assert.False(t, ts.IsSolo())
+		assert.Equal(t, 3, ts.Len())
+		assert.Equal(t, b1, ts.At(0))
+		assert.Equal(t, b2, ts.At(1))
+		assert.Equal(t, b3, ts.At(2))
+		assert.Equal(t, []*Block{b1, b2, b3}, ts.ToSlice())
+
+		tsParentWeight, _ := ts.ParentWeight()
+		assert.Equal(t, uint64(1337000), tsParentWeight)
+		assert.Equal(t, NewSortedCidSet(b1.Cid(), b2.Cid(), b3.Cid()), ts.ToSortedCidSet())
+		tsParents, _ := ts.Parents()
+		assert.Equal(t, b1.Parents, tsParents)
+		tsHeight, _ := ts.Height()
+		assert.Equal(t, uint64(b1.Height), tsHeight)
+		tsTicket, _ := ts.MinTicket()
+		assert.Equal(t, b1.Ticket, tsTicket)
+
+		assert.Equal(t, "{ "+b1.Cid().String()+" "+b2.Cid().String()+" "+b3.Cid().String()+" }", ts.String())
+	})
 
-	ts := RequireTestTipSet(t)
-	b1, b2, b3 := RequireTestBlocks(t)
+	t.Run("break ties with CID", func(t *testing.T) {
+		pq := uint64(1337000)
 
-	cidsExp := NewSortedCidSet(b1.Cid(), b2.Cid(), b3.Cid())
-	assert.Equal(t, cidsExp, ts.ToSortedCidSet())
-}
+		b1 := block(t, []byte{1}, 1, cid1, pq, "1")
+		b2 := block(t, []byte{1}, 1, cid1, pq, "2")
 
-func TestTipSetString(t *testing.T) {
-	tf.UnitTest(t)
+		ts := RequireNewTipSet(t, b1, b2)
+		if bytes.Compare(b1.Cid().Bytes(), b2.Cid().Bytes()) < 0 {
+			assert.Equal(t, []*Block{b1, b2}, ts.ToSlice())
+		} else {
+			assert.Equal(t, []*Block{b2, b1}, ts.ToSlice())
+		}
+	})
 
-	ts := RequireTestTipSet(t)
-	b1, b2, b3 := RequireTestBlocks(t)
+	t.Run("equality", func(t *testing.T) {
+		ts1a := RequireNewTipSet(t, b3, b2, b1)
+		ts1b := RequireNewTipSet(t, b1, b2, b3)
+		ts2 := RequireNewTipSet(t, b1, b2)
+		ts3 := RequireNewTipSet(t, b2)
+
+		assert.Equal(t, ts1a, ts1a)
+		assert.Equal(t, ts1a, ts1b)
+		assert.NotEqual(t, ts1a, ts2)
+		assert.NotEqual(t, ts1a, ts3)
+		assert.NotEqual(t, ts1a, NoTipSet)
+		assert.NotEqual(t, ts2, NoTipSet)
+		assert.NotEqual(t, ts3, NoTipSet)
+	})
 
-	cidsExp := NewSortedCidSet(b1.Cid(), b2.Cid(), b3.Cid())
-	strExp := cidsExp.String()
-	assert.Equal(t, strExp, ts.String())
-}
+	t.Run("slice", func(t *testing.T) {
+		ts := RequireNewTipSet(t, b3, b2, b1) // Presented in reverse order
+		slice := ts.ToSlice()
+		assert.Equal(t, []*Block{b1, b2, b3}, slice)
+		slice[1] = b1
+		slice[2] = b2
 
-func TestTipSetToSlice(t *testing.T) {
-	tf.UnitTest(t)
+		assert.NotEqual(t, slice, ts.ToSlice())
+		assert.Equal(t, []*Block{b1, b2, b3}, ts.ToSlice()) // tipset is immutable
+	})
 
-	ts := RequireTestTipSet(t)
-	b1, b2, b3 := RequireTestBlocks(t)
-	tips := []*Block{b1, b2, b3}
+	t.Run("empty", func(t *testing.T) {
```

Comment:
This and the following test descriptions should contain `NewTipSet` to clarify exactly what's being tested.